src/reactor/mod.rs

```diff
@@ -4,40 +4,37 @@
 //! happening in `tokio-core`. This reactor (or event loop) is used to run
 //! futures, schedule tasks, issue I/O requests, etc.
 
-use std::cell::RefCell;
-use std::cmp;
 use std::fmt;
-use std::io::{self, ErrorKind};
-use std::mem;
-use std::rc::{Rc, Weak};
-use std::sync::Arc;
-use std::sync::atomic::{AtomicUsize, ATOMIC_USIZE_INIT, Ordering};
-use std::time::{Instant, Duration};
+use std::io;
+use std::time::Duration;
 
-use futures::{Future, IntoFuture, Async};
+use futures::{Future, IntoFuture};
 use futures::future::{self, Executor, ExecuteError};
-use futures::executor::{self, Spawn, Notify};
-use futures::sync::mpsc;
-use futures::task::Task;
-use mio;
-use mio::event::Evented;
-use slab::Slab;
 
-use heap::{Heap, Slot};
+#[cfg(not(target_os = "fuchsia"))]
+mod mio;
+#[cfg(not(target_os = "fuchsia"))]
+use self::mio as sys;
 
-mod io_token;
-mod timeout_token;
+#[cfg(target_os = "fuchsia")]
+mod fuchsia;
+#[cfg(target_os = "fuchsia")]
+use self::fuchsia as sys;
+#[cfg(target_os = "fuchsia")]
+use std::sync::Arc;
 
-mod poll_evented;
 mod timeout;
 mod interval;
-pub use self::poll_evented::PollEvented;
+
+#[cfg(not(target_os = "fuchsia"))]
```

Comment:
This is the main change to the external interface-- Fuchsia will no longer offer `PollEvented`, but instead exposes the `PacketReceiver` trait which can be registered via a `Core`, `Handle` or `Remote`. After registration, an async message can be queued on the `zx::Port` with the key of the appropriate `ReceiverRegistration`, and the packet will be delivered. This allows for delivery of non-signal-type packets.